Windows self-path fails on Yggdrasil

It seems that [Yggdrasil Windows builder](https://github.com/JuliaPackaging/Yggdrasil/pull/6462) does not like #103.
 
```
[13:48:16] libblastrampoline.c: In function ‘DllMain’:
--
  | [13:48:16] libblastrampoline.c:404:1: error: parameter name omitted
  | [13:48:16]  BOOL APIENTRY DllMain(HINSTANCE hModule, DWORD code, void *) {
  | [13:48:16]  ^~~~
  | [13:48:16] make: *** [Makefile:55: build/libblastrampoline.o] Error 1
  | [13:48:16] make: *** Waiting for unfinished jobs....
  | [13:48:19]  ---> make -j${nproc} prefix=${prefix} install
  | [13:48:19]  ---> make -j${nproc} prefix=${prefix} install
  | [13:48:19] Previous command exited with 2
  | [13:48:19] Child Process exited, exit code 2
```

https://buildkite.com/julialang/yggdrasil/builds/1752#01872351-9817-4473-9196-3e177c0f85a1

--- a/src/libblastrampoline.c
+++ b/src/libblastrampoline.c
@@ -401,7 +401,7 @@ LBT_DLLEXPORT int32_t lbt_forward(const char * libname, int32_t clear, int32_t v
  */
 #ifdef _OS_WINDOWS_
 void * _win32_self_handle;
-BOOL APIENTRY DllMain(HINSTANCE hModule, DWORD code, void *) {
+BOOL APIENTRY DllMain(HINSTANCE hModule, DWORD code, void *reserved) {
     if (code == DLL_PROCESS_ATTACH) {
         _win32_self_handle = (void *)hModule;
     }

